### PR TITLE
First draft for TCP check docs page

### PR DESCRIPTION
**Pull request by sujaya-sys**
## Overview
* Developer documentation for TCP checks
* For TCP related updates of our existing documentation a separate PR will be created

## Open Questions

- [x] The "TCP check" menu item is shown at the very bottom. I'd like to move it below "API checks". Does anyone know how this can be done? The `weight` in `_index` does not seem to work here.

### CODE REVIEW

**Comment by pmallol** [comment]
> ## Open Questions
> * [ ]  The "TCP check" menu item is shown at the very bottom. I'd like to move it below "API checks". Does anyone know how this can be done? The `weight` in `_index` does not seem to work here.
> * [ ]  I commented out the tabs for CLI; TF; Pulumi in the code but they still show up in the preview. Any idea how to avoid that?

1. If I understand correctly we need to explicitly add it to the configuration file:
https://github.com/checkly/docs.checklyhq.com/blob/f77870f41c52645e692c5b92345653c82325550f/site/config.toml#L63-L82

2. In Hugo templating [you can comment out code](https://gohugo.io/content-management/shortcodes/#comment) using `{{ % comment %}}` or just deleting the content if not being used for the moment

File: site/content/docs/tcp-checks/_index.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+## Update or delete a TCP check
+
+{{< tabs "TCP check updating & deletion by interface" >}}
+{{< tab "UI" >}}
+<br>
+
+* **Update a TCP check:** To update a TCP check, click the kebab menu (three dots) next to the check on your [main dashboard](https://app.checklyhq.com) and select `Edit`, or click `Edit` in the top-right corner of the check's Reporting page.
+
+* **Delete a TCP check:** To delete a TCP check, use the kebab menu (three dots) on the [main dashboard](https://app.checklyhq.com) and select `Delete`.
```
Comment (pmallol) [inline site/content/docs/tcp-checks/_index.md:60]:
```suggestion
* **Delete a TCP check:** To delete a TCP check, use the kebab menu (three dots) on the [main dashboard](https://app.checklyhq.com) and select `Delete`, or click `Edit` in the top-right corner of the check's Reporting page and scroll down to the bottom section and click the Delete button.
```

**Review by pmallol — CHANGES_REQUESTED** [review]

**Comment by sujaya-sys** [comment]
> > ## Open Questions
> > 
> > * [ ]   The "TCP check" menu item is shown at the very bottom. I'd like to move it below "API checks". Does anyone know how this can be done? The `weight` in `_index` does not seem to work here.
> > * [ ]   I commented out the tabs for CLI; TF; Pulumi in the code but they still show up in the preview. Any idea how to avoid that?
> 
> cc @sujaya-sys
> 
> 1. If I understand correctly we need to explicitly add it to the configuration file:
>    https://github.com/checkly/docs.checklyhq.com/blob/f77870f41c52645e692c5b92345653c82325550f/site/config.toml#L63-L82
> 2. In Hugo templating [you can comment out code](https://gohugo.io/content-management/shortcodes/#comment) using `{{ % comment %}} [...] {{ % /comment %}}` or just deleting the content if not being used for the moment

Awesome, thank you @pmallol!

**Review by tnolet — COMMENTED** [review]
Looks like a good placeholder for further content, having the UI / CLI / TF / Pulumi in tabs feels a bit much. The reference documentation can be used here to explain the what and why of a TCP check. We can describe shortly how the MaC variants can be used to create one, but we should add the extensive reference docs for a TCPCheck construct for CLI/TF/Pulumi in the relevant sections. 

This means I would ditch the tabbed menus once we have these providers and just give a short example inline. This makes it easier to read as scrolling is much more a native reading experience than clicking tabs.

I would just make a small H2 or H3. This will also make them show up in the Table of Contents on the right side. Then we can link to the "extensive" reference docs in the relevant section, e.g. CLI / Constructs.

But lets fix that once we have actual MaC providers.

**Comment by sujaya-sys** [comment]
> Looks like a good placeholder for further content, having the UI / CLI / TF / Pulumi in tabs feels a bit much. The reference documentation can be used here to explain the what and why of a TCP check. We can describe shortly how the MaC variants can be used to create one, but we should add the extensive reference docs for a TCPCheck construct for CLI/TF/Pulumi in the relevant sections.
> 
> This means I would ditch the tabbed menus once we have these providers and just give a short example inline. This makes it easier to read as scrolling is much more a native reading experience than clicking tabs.
> 
> I would just make a small H2 or H3. This will also make them show up in the Table of Contents on the right side. Then we can link to the "extensive" reference docs in the relevant section, e.g. CLI / Constructs.
> 
> But lets fix that once we have actual MaC providers.

That makes a lot of sense, thanks Tim! Let's keep the tabs to make it clear for early users that we're only supporting the UI ATM and replace them with proper references once CLI/TF etc. docs are available 👍

**Review by pmallol — APPROVED** [review]

**Pull request merged**